### DISCUSSION TITLE
What’s the reasoning behind the -> to denote type annotations in the SDL?

### DISCUSSION THREAD

**Discussion by jonaird** [geldata/gel#2200] [General]
It seems a bit unusual and also a bit awkward to type. Something like the way Java or Typescript does it might be a bit more natural? Is it to enforce a type on all fields?

**Comment by elprans** [comment]
SDL syntax is essentially abbreviated DDL (with verbs stripped).  We use `->` to annotate the result type of a schema construct.  Functions, for example, also use `->` to specify the return type.  Historically, we chose `->`, because `:` was conflicting, and because `->` also resembles a graph link, i.e `type Foo { link bar -> Bar }`.
